pkg/infra/game/api/api.go

```diff
@@ -0,0 +1,34 @@
+package api
+
+import (
+	"infra/game/agent"
+	"infra/game/decision"
+	"infra/game/stage/fight"
+	"infra/game/stage/loot"
+	"infra/game/state"
+
+	"github.com/benbjohnson/immutable"
+
+	t0 "infra/teams/team0"
+)
+
+var mode = "default"
+
+// TODO: Change to using views
+func AllocateLoot(globalState state.State, weaponLoot []uint, shieldLoot []uint) (allocatedState state.State) {
+	switch mode {
+	case "0":
+		return t0.AllocateLoot(globalState, weaponLoot, shieldLoot)
+	default:
+		return loot.AllocateLoot(globalState, weaponLoot, shieldLoot)
+	}
+
+}
+func AgentFightDecisions(state *state.View, agents map[string]agent.Agent, previousDecisions *immutable.Map[string, decision.FightAction]) map[string]decision.FightAction {
```

Comment:
Is this this public facing API for agent teams, I thought we were providing the `Strategy` interface for the API? Unless you believe agent teams should be able to deviate from decisions chosen by the agents themselves?

Reply:
I think teams can still use the same strategy interface, though if they want they could change the way agents communicate with one another. If for example, agents are allowed to steal resources it may be easier to code here. In the main version I am assuming if no consensus is met no loot would be allocated to anyone, but this function might override that behavior and allow some agents to steal loot while others would not get any.

Reply:
Oh wait I hadn't realised you were referring to this section of code. I'll pass the map of agents to this function as well.